### PR TITLE
RHTAP-1134: Support nested Gitlab repos

### PR DESCRIPTION
**Pull request by gbenhaim**
Gitlab can have nested repositories, thus the parsing of the repo url should handle it.

### CODE REVIEW

File: pkg/git/gitlab/gitlab_client.go [hunk 1]
```diff
@@ -39,7 +39,11 @@ type GitlabClient struct {
 // Returns the merge request web URL.
 // If there is no error and web URL is empty, it means that the merge request is not needed (main branch is up to date).
 func (g *GitlabClient) EnsurePaCMergeRequest(repoUrl string, d *gp.MergeRequestData) (webUrl string, err error) {
-	projectPath := getProjectPathFromRepoUrl(repoUrl)
+	projectPath, err := getProjectPathFromRepoUrl(repoUrl)
+
```
Comment (mmorhun) [inline pkg/git/gitlab/gitlab_client.go:43]: nitpick: according to the codestyle, extra new line is not needed.

**Review by mmorhun — APPROVED** [review]

File: pkg/git/gitlab/gitlab_client.go (hunk 1 above)
Comment (gbenhaim) [reply pkg/git/gitlab/gitlab_client.go]: fixed

**Review by mmorhun — APPROVED** [review]

File: pkg/git/gitlab/gitlab_helper_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		{
+			"https://gitlab.com/redhat/rhel/containers/rhtap-prototype.git",
+			"redhat/rhel/containers/rhtap-prototype",
+			nil,
+		},
+		{
+			"",
+			"",
+			nil,
+		},
```
[the remark this answers is not in the document]
Comment (gbenhaim) [reply pkg/git/gitlab/gitlab_helper_test.go:38]: added

**Comment by gbenhaim** [comment]
@chmeliik is there a bot that will merge it? or someone need to click the button?

**Comment by chmeliik** [comment]
> @chmeliik is there a bot that will merge it? or someone need to click the button?

It's a button-clicking kind of repo :slightly_smiling_face: 

I've re-started the envtest that failed on uploading coverage data

**Comment by gbenhaim** [comment]
@adambkaplan @mmorhun who can click the merge button? btw the envtest that fails doesn't seems to be related to my change, maybe there is an issue with the test.

**Comment by chmeliik** [comment]
Will try re-running the envtest to see if it's intermittent. The relevant part of the logs for future reference:

<details>

```
• [FAILED] [16.331 seconds]
Component initial build controller Test Pipelines as Code build preparation [It] should reuse the same webhook secret for multi component repository
/home/runner/work/build-service/build-service/controllers/component_build_controller_test.go:483

  Timeline >>
  2023-12-14T15:00:01Z	ERROR	GitTektonResourcesRenovator	not found Pipelines as Code secret in %s namespace: %w	{"controller": "buildpipelineselector", "controllerGroup": "appstudio.redhat.com", "controllerKind": "BuildPipelineSelector", "BuildPipelineSelector": {"name":"build-pipeline-selector","namespace":"build-service"}, "namespace": "build-service", "name": "build-pipeline-selector", "reconcileID": "3772380a-c3a1-4fa4-b7ef-f05e0d44e6ad", "build-service": "Secret \"pipelines-as-code-secret\" not found", "action": "VIEW", "error": "Secret \"pipelines-as-code-secret\" not found"}
  github.com/redhat-appstudio/build-service/controllers.(*GitTektonResourcesRenovater).Reconcile
  	/home/runner/work/build-service/build-service/controllers/git_tekton_resources_renovater.go:119
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:119
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:316
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227
  2023-12-14T15:00:01Z	DEBUG	events	Secret "pipelines-as-code-secret" not found	{"type": "Warning", "object": {"kind":"Secret","apiVersion":"v1"}, "reason": "ErrorReadingPaCSecret"}
  2023-12-14T15:00:01Z	INFO	ComponentOnboarding	Waiting for devfile model in component	{"controller": "component", "controllerGroup": "appstudio.redhat.com", "controllerKind": "Component", "Component": {"name":"test-component-pacprep","namespace":"default"}, "namespace": "default", "name": "test-component-pacprep", "reconcileID": "687aa1f7-bf92-40fe-90d0-68506376a5fc", "ComponentGitSource": {"url":"https://github.com/devfile-samples/devfile-sample-java-springboot-basic-test-component-pacprep","revision":"main"}}
  2023-12-14T15:00:01Z	INFO	ComponentOnboarding.PaC-setup	Starting Pipelines as Code provision for the Component	{"controller": "component", "controllerGroup": "appstudio.redhat.com", "controllerKind": "Component", "Component": {"name":"test-component-pacprep","namespace":"default"}, "namespace": "default", "name": "test-component-pacprep", "reconcileID": "fe160618-e727-4239-b1b6-86aa60220333", "ComponentGitSource": {"url":"https://github.com/devfile-samples/devfile-sample-java-springboot-basic-test-component-pacprep","revision":"main"}}
  2023-12-14T15:00:01Z	ERROR	ComponentOnboarding.PaC-setup	failed to create webhooks secrets secret	{"controller": "component", "controllerGroup": "appstudio.redhat.com", "controllerKind": "Component", "Component": {"name":"test-component-pacprep","namespace":"default"}, "namespace": "default", "name": "test-component-pacprep", "reconcileID": "fe160618-e727-4239-b1b6-86aa60220333", "ComponentGitSource": {"url":"https://github.com/devfile-samples/devfile-sample-java-springboot-basic-test-component-pacprep","revision":"main"}, "action": "ADD", "error": "secrets \"pipelines-as-code-webhooks-secret\" already exists"}
  github.com/redhat-appstudio/build-service/controllers.(*ComponentBuildReconciler).ensureWebhookSecret
  	/home/runner/work/build-service/build-service/controllers/component_build_controller_pac.go:533
  github.com/redhat-appstudio/build-service/controllers.(*ComponentBuildReconciler).ensureWebhookSecret
  	/home/runner/work/build-service/build-service/controllers/component_build_controller_pac.go:536
  github.com/redhat-appstudio/build-service/controllers.(*ComponentBuildReconciler).ProvisionPaCForComponent
  	/home/runner/work/build-service/build-service/controllers/component_build_controller_pac.go:124
  github.com/redhat-appstudio/build-service/controllers.(*ComponentBuildReconciler).Reconcile
  	/home/runner/work/build-service/build-service/controllers/component_build_controller.go:472
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:119
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:316
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227
  2023-12-14T15:00:01Z	ERROR	ComponentOnboarding	Pipelines as Code provision transient error	{"controller": "component", "controllerGroup": "appstudio.redhat.com", "controllerKind": "Component", "Component": {"name":"test-component-pacprep","namespace":"default"}, "namespace": "default", "name": "test-component-pacprep", "reconcileID": "fe160618-e727-4239-b1b6-86aa60220333", "ComponentGitSource": {"url":"https://github.com/devfile-samples/devfile-sample-java-springboot-basic-test-component-pacprep","revision":"main"}, "error": "secrets \"pipelines-as-code-webhooks-secret\" already exists"}
  github.com/redhat-appstudio/build-service/controllers.(*ComponentBuildReconciler).Reconcile
  	/home/runner/work/build-service/build-service/controllers/component_build_controller.go:480
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:119
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:316
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227
  2023-12-14T15:00:01Z	ERROR	Reconciler error	{"controller": "component", "controllerGroup": "appstudio.redhat.com", "controllerKind": "Component", "Component": {"name":"test-component-pacprep","namespace":"default"}, "namespace": "default", "name": "test-component-pacprep", "reconcileID": "fe160618-e727-4239-b1b6-86aa60220333", "error": "secrets \"pipelines-as-code-webhooks-secret\" already exists"}
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:329
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227
```

</details>

**Pull request merged**